be/src/storage/data_dir.cpp

```diff
@@ -464,6 +465,82 @@ void DataDir::perform_path_gc_by_tablet() {
     LOG(INFO) << "finished one time path gc by tablet.";
 }
 
+bool DataDir::_need_gc_delta_column_files(
+        const std::string& path, int64_t tablet_id,
+        std::unordered_map<int64_t, std::unordered_set<std::string>>& delta_column_files) {
+    if (delta_column_files.count(tablet_id) == 0) {
```

Comment:
```suggestion
    if (tablet->keys_type() != KeysType::PRIMARY_KEYS) {
      return false;
    }
    TabletSharedPtr tablet = _tablet_manager->get_tablet(tablet_id, false);
    if (tablet != nullptr && tablet->tablet_state() == TABLET_RUNNING) [
      return false;
    }
    if (delta_column_files.count(tablet_id) == 0) {
      DeltaColumnGroupList dcgs;
      if (tablet->updates()->need_apply()) {
        // if this tablet has apply task to handle, skip it, because it can't get latest dcgs
        return false;
      }
      auto st = TabletMetaManager::scan_tablet_delta_column_group(_kv_store, tablet_id, &dcgs);
      if (!st.ok()) {
        LOG(WARNING) << "scan tablet delta column group failed, tablet_id: " << tablet_id << ", st: " << st;
         return false;
      }
      auto& files = delta_column_files[tablet_id];
      for (const auto& dcg : dcgs) {
         const auto& column_files = dcg->relative_column_files();
         files.insert(column_files.begin(), column_files.end());
      }
    }
    std::string filename = std::filesystem::path(path).filename().string();
    return delta_column_files[tablet_id].count(filename) == 0;
```